roles/sap_ha_install_hana_hsr/tasks/configure_hsr.yml

```diff
@@ -1,61 +1,68 @@
 ---
-
-# [1] Enable HANA System Replication
-- name: SAP HSR - [1] Enable HANA System Replication
+- name: "SAP HSR - Check System Replication Status"
   shell: |
-      source /usr/sap/{{ sap_ha_install_hana_hsr_sid }}/home/.sapenv.sh && \
-      /usr/sap/{{ sap_ha_install_hana_hsr_sid | upper }}/HDB{{ sap_ha_install_hana_hsr_instance_number }}/exe/hdbnsutil \
-      -sr_enable --name="{{ sap_ha_install_hana_hsr_alias }}"
+    source /usr/sap/{{ sap_ha_install_hana_hsr_sid }}/home/.sapenv.sh && \
+    /usr/sap/{{ sap_ha_install_hana_hsr_sid | upper }}/HDB{{ sap_ha_install_hana_hsr_instance_number }}/exe/hdbnsutil \
+    -sr_state
   args:
     executable: /bin/bash
   become: true
   become_user: "{{ sap_ha_install_hana_hsr_sid | lower }}adm"
-  register: enablesr
-  changed_when: "'successfully enabled system as system replication source site' in enablesr.stdout"
-  when:
-    - sap_ha_install_hana_hsr_role == 'primary'
+  register: __sap_ha_install_hana_hsr_srstate
+  changed_when: false
+  failed_when: false
 
-# [2] Stop HANA instance
-- name: SAP HSR - [2] Stop HANA instance
+- name: "SAP HSR - Enable HANA System Replication on primary node"
   shell: |
-      /usr/sap/{{ sap_ha_install_hana_hsr_sid | upper }}/HDB{{ sap_ha_install_hana_hsr_instance_number }}/HDB stop
+    source /usr/sap/{{ sap_ha_install_hana_hsr_sid }}/home/.sapenv.sh && \
+    /usr/sap/{{ sap_ha_install_hana_hsr_sid | upper }}/HDB{{ sap_ha_install_hana_hsr_instance_number }}/exe/hdbnsutil \
+    -sr_enable --name="{{ item.hana_site }}"
   args:
     executable: /bin/bash
   become: true
   become_user: "{{ sap_ha_install_hana_hsr_sid | lower }}adm"
-  register: stopinstance
-  changed_when: "'StopSystem' in stopinstance.stdout"
+  register: enablesr
+  changed_when: "'successfully enabled system as system replication source site' in enablesr.stdout"
   when:
-    - sap_ha_install_hana_hsr_role == 'secondary'
+    - ansible_hostname == item.node_name
```

Comment:
isn't this condition always true?

Reply:
I would add a description above the task regarding that loop key then, the logic makes sense when you explain it

Reply:
Reviewed. Going through the loop is required because the cluster definition list contains the site name which is needed to configure HSR. The conditionals identify the matching target node in the definition and the task executes with the node individual value.

Reply:
@kksat I am now resolving this conversation after @ja9fuchs replied and explained it. I also see that you approved the PR already. Feel free to open an issue related to your comment.